osf/models/user.py

```diff
@@ -1057,6 +1057,7 @@ def create(cls, username, password, fullname, accepted_terms_of_service=None):
         )
         user.update_guessed_names()
         user.set_password(password)
+        user.save()
```

Comment:
Is this related to the relationship not saved issue you mentioned? Can you elaborate a little bit more in the PR description with references. To me, this seems to be an important change that we and future devs needs to be aware of. Thanks!

Reply:
It is related indeed, I made a change here, not to burden everyone to save user before making relations